neo4j/tests/test_neo4j.py

```diff
@@ -1,40 +1,90 @@
-# (C) Datadog, Inc. 2010-2016
-# All rights reserved
-# Licensed under Simplified BSD License (see LICENSE)
-
 import pytest
 
-from datadog_checks.errors import CheckException
-from datadog_checks.neo4j import Neo4jCheck
-
-from .common import CHECK_NAME, CONNECTION_FAILURE, NEO4J_MINIMAL_CONFIG, NEO4J_VARS
+from datadog_checks.base import ConfigurationError
+from datadog_checks.neo4j import GLOBAL_DB_NAME, NAMESPACE, Config, Neo4jCheck
 
 
+@pytest.mark.integration
 @pytest.mark.usefixtures('dd_environment')
-def test_minimal_config(aggregator):
-    c = Neo4jCheck(CHECK_NAME, {}, {})
-    c.check(NEO4J_MINIMAL_CONFIG)
+def test_service_check(aggregator, instance):
+    check = Neo4jCheck('neo4j', {}, {})
 
-    # Test service check
-    aggregator.assert_service_check('neo4j.can_connect', status=Neo4jCheck.OK)
+    instance['neo4j_dbs'] = ['neo4j', 'system']
 
-    # Test metrics
-    testable_metrics = NEO4J_VARS
+    check.check(instance)
 
-    for mname in testable_metrics:
-        aggregator.assert_metric('neo4j.{}'.format(mname), tags=[])
+    print(aggregator.metric_names)
 
-    aggregator.assert_all_metrics_covered()
+    if instance.get('neo4j_version') == '3.5':
+        aggregator.assert_metric(
+            name='{}.page_cache.hits'.format(NAMESPACE), tags=['db_name:{}'.format(GLOBAL_DB_NAME)],
+        )
+        aggregator.assert_metric(
+            name='{}.check_point.events'.format(NAMESPACE), tags=['db_name:{}'.format(GLOBAL_DB_NAME)],
+        )
+    elif instance.get('neo4j_version') == '4.0':
+        aggregator.assert_metric(
+            name='{}.page_cache.hits'.format(NAMESPACE), tags=['db_name:{}'.format(GLOBAL_DB_NAME)],
+        )
+        aggregator.assert_metric(name='{}.check_point.events'.format(NAMESPACE), tags=['db_name:neo4j'])
+        aggregator.assert_metric(name='{}.check_point.events'.format(NAMESPACE), tags=['db_name:system'])
+    else:
```

Comment:
Should this include a test for `4.1` as well?

Reply:
4.1 is not public yet :)

Reply:
Yes, but it is in the code now so should it be removed from the code too then?

Reply:
Yeah you are right, that might need fixing before we create a PR against the upstream repository. Good find!

Reply:
Does it make a difference if we remove it before createing a PR against the upstream repo, isn't this repo public anyway?